src/UnigramWordPredictor.java

```diff
@@ -101,7 +127,24 @@ public void train(Scanner scanner) {
   public String predictNextWord(List<String> context) {
     // TODO: Return a predicted word given the words preceding it
     // Hint: only the last word in context should be looked at
-    return null;
+
+    // index of last word
+    int index = context.size()-1;
+
+    // identify the last word in the current contest
+    String word = context.get(index);
+
+    // return the List<String> containing all the possible next words
+    List<String> possibleNexts = neighborMap.get(word);
+
+    // randomly choose a word from the list
+    Random choose = new Random();
+    int numberOfPossibilities = possibleNexts.size();
+    int randomIndex = choose.nextInt(numberOfPossibilities);
+    String nextWord = possibleNexts.get(randomIndex);
+
+    // return a random word
+    return nextWord;
```

Comment:
This works great! But what would happen if we haven't seen the word in the context before or if it was the last word in the file and duplicated nowhere else? What would your code currently do? What options would you have to deal with this?